cirq/experiments/n_qubit_tomography.py

```diff
@@ -0,0 +1,194 @@
+            rotation sequence and bit string outcome for that rotation sequence.
+        """
+        num_rots = len(self.rot_sweep)
+        num_states = 2**self.num_qubits
+
+        # Unitary matrices of each rotation circuit.
+        Us = np.array([
+            cirq.unitary(cirq.resolve_parameters(self.rot_moment, rots))
+            for rots in self.rot_sweep
+        ])
+        mat = np.einsum('jkm,jkn->jkmn', Us, Us.conj())
+        return mat.reshape((num_rots * num_states, num_states * num_states))
+
+    def fit_density_matrix(
+            self,
+            counts: np.ndarray,
+    ) -> 'TomographyResult':
+        """Solves equation mat * rho = probs.
+
+        Args:.
+            counts:  A 2D array where each row contains measured counts
+                of all n-qubit bitstrings for the corresponding pre-rotations
+                in rot_sweep.  The order of the probabilities corresponds to
+                to rot_sweep and the order of the bit strings corresponds to
+                increasing integers up to 2**(num_qubits)-1
+
+        Returns:
+            TomographyResult with density matrix corresponding to solution of
+            this system.
+        """
+        # normalize the input.
+        probs = counts / np.sum(counts, axis=1)[:, np.newaxis]
+        # use least squares to get solution.
+        c, _, _, _ = np.linalg.lstsq(self.mat, np.asarray(probs).flat, rcond=-1)
+        rho = c.reshape((2**self.num_qubits, 2**self.num_qubits))
+        return TomographyResult(rho)
+
+
+def state_tomography(
+        sampler: cirq.Sampler,
+        qubits: List[cirq.Qid],
+        circuit: cirq.Circuit,
+        repetitions: int = 1000,
+        prerotations: List[Tuple[float, float]] = None,
+) -> 'TomographyResult':
+    """This performs n qubit tomography on a cirq circuit
+
+    Follows https://web.physics.ucsb.edu/~martinisgroup/theses/Neeley2010b.pdf
+    A.1. State Tomography.
+    This is a high level interface for StateTomographyExperiment.
+    Args:
+        circuit: circuit to do the tomography on.
+        qubits: qubits to do the tomography on.
+        sampler: sampler to collect the data from.
+        repetitions: number of times to sample each rotations
+        prerotations: Tuples of (phase_exponent, exponent) parameters for gates
+            to apply to the qubits before measurement. The actual rotation
+            applied will be
+                cirq.PhasedXPowGate(
+                    phase_exponent=phase_exponent, exponent=exponent)
+            If none, we use [(0, 0), (0, 0.5), (0.5, 0.5)], which corresponds to
+            rotation gates [I, X**0.5, Y**0.5]
+    Returns:
+        Tomography result which contains the density matrix of the qubits
+        determined by tomography.
+    """
+    if prerotations == None:
+        prerotations = [(0, 0), (0, 0.5), (0.5, 0.5)]
+
+    exp = StateTomographyExperiment(qubits, prerotations)
+    rot_moment, rot_sweep = (exp.rot_moment, exp.rot_sweep)
```

Comment:
Nit: Don't need these parentheses.

These probably don't need to be stored in local variables.